onnxruntime/core/providers/cuda/tensor/expand.cc

```diff
@@ -69,7 +69,13 @@ Status Expand::ComputeInternal(OpKernelContext* ctx) const {
 
   // new shape to be expanded to
   const auto* p_shape = input_shape_tensor.template Data<int64_t>();
-  std::vector<int64_t> output_dims{p_shape, p_shape + input_shape_tensor.Shape().Size()};
+  std::vector<int64_t> output_dims(input_shape_tensor.Shape().Size());
+  for (auto i = 0; i < input_shape_tensor.Shape().Size(); i++) {
+    // According to the ONNX spec: "Dimensions are right alignment;
+    // Two corresponding dimension must have the same value, or one of them is equal to 1",
+    // value -1 from PyTorch has the same meaning of 1 here.
+    output_dims[i] = p_shape[i] == -1 ? 1 : p_shape[i];
```

Comment:
>p_shape[i] == -1 [](start = 21, length = 16)

Can you check with Lara to see if this is a expected behavior of pytorch converter?


According to the spec
"Two corresponding dimension must have the same value, or one of them is equal to 1“

-1 doesn't looks like a legal value here

Reply:
Roll back this change. Need to check if exporter can handle this. Before they can fix, we can add some temporary transformer script for this.